src/stan/error_handling/matrix/check_size_match.hpp

```diff
@@ -16,14 +32,14 @@ namespace stan {
                                  const std::string& name_j, 
                                  T_size2 j) {
       typedef typename boost::common_type<T_size1,T_size2>::type common_type;
-      if (static_cast<common_type>(i) == static_cast<common_type>(j))
+      if (likely(static_cast<common_type>(i) == static_cast<common_type>(j)))
```

Comment:
When do we have this cast to a common type where `==` itself wouldn't be defined?  That is, can we just get rid of this cast?